README.md

```diff
@@ -69,4 +79,60 @@ $ curl localhost:8080/metrics
 $ docker build . -t prometheus-sample-app
 $ docker run -it -p 8080:8080 prometheus-sample-app /bin/main -listen_address=0.0.0.0:8080 generate -metric_type=all -is_random=true
 $ curl localhost:8080/metrics
-```
\ No newline at end of file
+```
+
+## Clustering:
+Deploy the example deployment configuration of 5 instances of Prometheus-Sample-App along with configured OTEL Collector.
+    
+### Pre-requisites:
+- Docker
+- Docker Image Prometheus-Sample-App
+- Minikube
+
+### Deployment on Kubernetes:
+- Run Docker
+- Start Minikube
+  ```bash
+    $ minikube start
+  ```
+- Start Minikube dashboard
+  ```bash
+    $ minikube dashboard
+  ```
+- Run following command to deploy: 
+    ```bash
+    $ kubectl apply -f otel-collector-k8s-deployment.yaml
+    $ kubectl create clusterrolebinding service-reader-pod --clusterrole=service-reader --serviceaccount=default:default
+    $ kubectl apply -f prometheus-sample-app-k8s-deployment.yaml
+    ```
+- Run following command to monitor logs from OTEL Collector Logging exporter :
+    ```bash
+    $ kubectl logs <otel-collector-pod-name>
+    ```
+### Deployment on EKS:
+- Create your cluster on EKS
+  ```bash
+  $ eksctl create cluster --name <cluster-name> --region <region> --with-oidc --ssh-access --ssh-public-key <public-key>
+  ```
+- Create repository on Amazon ECR to push docker image of Prometheus-Sample-App
+- Push the prometheus_sample_app docker image to this repository
+    ```bash
+    $ aws ecr get-login-password --region region | docker login --username AWS --password-stdin aws_account_id.dkr.ecr.region.amazonaws.com
+    $ docker build -t prometheus_sample_app .
+    $ docker tag prometheus_sample_app:latest aws_account_id.dkr.ecr.region.amazonaws.com/my-repository:tag
+    $ docker push aws_account_id.dkr.ecr.region.amazonaws.com/my-repository:tag
+    ```
+- Update imagePullPolicy of 'prometheus-sample-app-k8s-deployment.yaml' to IfNotPresent
+- Copy image URI from the AWS ECR repository and update in 'prometheus-sample-app-k8s-deployment.yaml'
+- Run following command to deploy:
+    ```bash
+    $ kubectl apply -f otel-collector-k8s-deployment.yaml
+    $ kubectl create clusterrolebinding service-reader-pod --clusterrole=service-reader --serviceaccount=default:default
+    $ kubectl apply -f prometheus-sample-app-k8s-deployment.yaml
+    ```
+- Run following command to monitor logs from OTEL Collector Logging exporter :
+    ```bash
+    $ kubectl logs <otel-collector-pod-name>
+    ```
+
+Currently, OTEL Collector is configured with Logging exporter. Since all 5 replica Prometheus-Sample-App pods will produce identical metric names and labels, so Prometheus Exporter should be configured with additional details to ensure distinct metrics identity. 
```

Comment:
Is there any information we can include here regarding what sort of additional details need to be included and how to include them?

Reply:
@Aneurysm9 I am unsure right now what can be done, my understanding is to generate unique source labels through configuration. But I am not entirely sure. So I have updated the comment.